quantum/util/quantum_sequencer_lite.h

```
@@ -0,0 +1,299 @@
+#define BLOOMBERG_QUANTUM_SEQUENCER_LITE_H
+
+#if __cplusplus >= 201402L
+#define BLOOMBERG_QUANTUM_SEQUENCER_LITE_SUPPORT
+#endif // __cplusplus >= 201402L
+
+#ifdef BLOOMBERG_QUANTUM_SEQUENCER_LITE_SUPPORT
+
+#include <quantum/quantum_dispatcher.h>
+#include <quantum/interface/quantum_iqueue.h>
+#include <quantum/interface/quantum_ithread_context_base.h>
+#include <quantum/util/quantum_sequencer_lite_configuration.h>
+#include <quantum/util/quantum_sequence_key_statistics.h>
+#include <vector>
+#include <unordered_map>
+#include <mutex>
+
+namespace Bloomberg {
+namespace quantum {
+
+//==============================================================================================
+//                                      class SequencerLite
+//==============================================================================================
+/// @class SequencerLite.
+/// @brief Implementation of a key-based task sequencing with quantum.
+/// @tparam SequenceKey Type of the key based that sequenced tasks are associated with
+/// @tparam Hash Hash-function used for storing instances of SequenceKey in hash maps
+/// @tparam KeyEqual The equal-function used for storing instances of SequenceKey in hash maps
+/// @tparam Allocator The allocator used for storing instances of SequenceKey in hash maps
+/// @note The interfaces of quantum::SequencerLite and quantum::Sequencer are the same. The major difference
+/// between their functionalities is that quantum::SequencerLite does not rely on quantum::Dispatcher to order tasks 
+/// based on their interdependence. In contrast, quantum::SequencerLite manages task ordering itself 
+/// by construcing a DAG of pending tasks. In quantum::SequencerLite, a task is pushed into quantum::Dispatcher 
+/// only when it's ready to be executed (i.e. when it has no pending dependents). This typically results 
+/// in executing scheduled tasks faster (w.r.t. quantum::Dispatcher) and wasting fewer CPU cycles in 
+/// quantum::Dispatcher.
```

Let's document the exception callback here; it's existence is buried within the configuration and barely documented there. Just a brief on the fact that the sequencer will swallow exceptions and call that callback if specified should be good. Might be good to copy that note to the old sequencer as well.